rewrite-java/src/main/java/org/openrewrite/java/tree/MethodTypeBuilder.java

```diff
@@ -32,36 +54,92 @@
     String name;
     List<Parameter> parameters = new ArrayList<>();
 
+    /**
+     * See {@link MethodTypeBuilder} for usage.
+     *
+     * @return A new builder.
+     */
     public static MethodTypeBuilder methodType() {
         return new MethodTypeBuilder();
     }
 
+    /**
+     * The declaring class is a fully qualified name of the receiver for this method type.
+     *
+     * @param fullyQualifiedClassName The fully-qualified name of the class on which this method is defined.
+     * @return this
```

Comment:
The `@return this` bits of all of these comments seem useless to me. 
We're not publishing, and do not plan to publish, our JavaDoc comments anywhere. They'll be seen by programmers looking at code in IDEs. There's no need for them to restate things that are obvious from the method signature, or in this case, obvious from the very common builder pattern. 

Every line of code and comment exacts a cognitive toll on the developer who reads it. 
A comment is a promise to the reader that if they invest their finite focus into comprehending it, that dividends will be paid later in the form of reduced cognitive burden to understand the code. 

A comment that isn't any easier to understand than the code it describes is a tiny broken promise.

Reply:
I am on board with that, this is a bad habit I picked up because of compiler warnings generated in Eclipse.